src/components/one-time-donation/components/ThirdStep.tsx

```diff
@@ -0,0 +1,137 @@
+import { Collapse, Grid, List, Divider, Typography, Button } from '@mui/material'
+import { createStyles, makeStyles } from '@mui/styles'
+import RadioGroupFormik from './RadioGroupFormik'
+import { useField } from 'formik'
+import { UseQueryResult } from 'react-query'
+import { useRouter } from 'next/router'
+import { routes } from 'common/routes'
+import { CampaignResponse } from 'gql/campaigns'
+import { useViewCampaign } from 'common/hooks/campaigns'
+import { CopyTextButton } from '../../common/CopyTextButton'
+import { useTranslation } from 'next-i18next'
+
+const useStyles = makeStyles(() =>
+  createStyles({
+    h2: {
+      fontFamily: 'Montserrat',
+      fontStyle: 'normal',
+      fontWeight: '500',
+      fontSize: '35px',
+      lineHeight: '120%',
+      marginTop: '58px',
+      marginBottom: '59px',
+      letterSpacing: '-0.5px',
+      color: '#343434',
+    },
+    divider: {
+      border: '1px solid #000000',
+    },
+    uncheked: {
+      width: '284px',
+      height: '75px',
+      background: '#FFFFFF',
+      border: '1px solid #000000',
+      boxSizing: 'border-box',
+      borderRadius: '37.5px',
+      marginBottom: '33px',
+    },
+    checked: {
+      width: '284px',
+      height: '75px',
+      background: '#D2F0FF',
+      border: '1px solid #000000',
+      boxSizing: 'border-box',
+      borderRadius: '37.5px',
+      marginBottom: '33px',
+    },
+    body: {
+      maxWidth: '657px',
+      marginLeft: 'auto',
+      marginRight: 'auto',
+    },
+  }),
+)
+
+export default function ThirdStep() {
+  const classes = useStyles()
+  const [field] = useField('payment')
+  const router = useRouter()
+  const slug = String(router.query.slug)
```

Comment:
Shouldn't router.query.slug be a string already?

Reply:
The router type system of parsed query string is typed as `string | string[]` as the query can be `?name=one` or `?name[]=one&name[]=two`

The funny thing is this can never happen on the page where we've got `campaign/[slug].tsx` but the router type still remains. If Next.js didn't use `router.query` for the pathname params this wouldn't be the case

---

I see another bigger problem here than the types.

We don't really know if the component `ThirdStep` would be used on a page which has variable param `[slug].tsx` in it's path. So? So it would break if this component is used on another page - `router.query.slug` would end up empty.

The best way to handle this is to pass the slug down the stream from the page component.